ops/terrad/terrad.go

```diff
@@ -353,28 +371,162 @@ func (t Terrad) InitOCR(keys []opsChainlink.NodeKeys) error {
 		transmitterArray = append(transmitterArray, string(transmitters[i]))
 	}
 
-	tx := SetConfig{
-		SetConfig: SetConfigDetails{
-			Signers:               signerArray,
-			Transmitters:          transmitterArray,
-			F:                     f,
-			OnchainConfig:         onchainConfig,
+	status = utils.LogStatus("InitOCR: begin proposal")
+	resp, err := t.executeOCR2(BeginProposal)
+	if err == nil {
+		if resp.Code != 0 {
+			err = fmt.Errorf("tx response contains error: %s %d", resp.Codespace, resp.Code)
+		} else if len(resp.Logs) == 0 {
+			err = errors.New("begin proposal produced no logs")
+		}
+	}
+	if status.Check(err) != nil {
+		return err
+	}
+
+	var id string
+	for _, e := range resp.Logs[0].Events {
+		if e.Type == "wasm" {
+			for _, a := range e.Attributes {
+				if a.Key == "proposal_id" {
+					if a.Value == "" {
+						return errors.New("empty proposal id")
+					}
+					id = a.Value
+				}
+			}
+		}
+	}
+	if id == "" {
+		return errors.New("failed to find event with attribute: wasm.proposal-id")
+	}
+
+	defer func() {
+		if rerr == nil {
+			return // Success
+		}
+		// Clean up
+		status = utils.LogStatus("InitOCR: clear proposal: " + id)
+		resp, err = t.executeOCR2(ClearProposal{
+			ClearProposal: ClearProposalDetails{ID: id},
+		})
+		if err == nil {
+			if resp.Code != 0 {
+				err = fmt.Errorf("tx response contains error: %s %d", resp.Codespace, resp.Code)
+			}
+		}
+		if status.Check(err) != nil {
+			fmt.Println(err)
+			return
+		}
+	}()
+
+	payees := make([]string, 0)
+	for i := 0; i < len(transmitterArray); i++ {
+		payees = append(payees, t.addr)
+	}
+	status = utils.LogStatus("InitOCR: propose config " + id)
+	resp, err = t.executeOCR2(ProposeConfig{
+		ProposeConfig: ProposeConfigDetails{
+			ID:            id,
+			Payees:        payees,
+			Signers:       signerArray,
+			Transmitters:  transmitterArray,
+			F:             f,
+			OnchainConfig: onchainConfig,
+		},
+	})
+	if err == nil && resp.Code != 0 {
+		err = fmt.Errorf("tx response contains error: %s %d", resp.Codespace, resp.Code)
+	}
+	if status.Check(err) != nil {
+		return err
+	}
```

Comment:
Potentially we could shed some repeated code by extracting this as:

```suggestion
	if err = t.Check(err); err != nil {
		return err
	}
```

Reply:
:+1: although I think pushing it inside of the execute method is even simpler.